File: ChineseAuctionAngular/ChineseAuctionAngular/src/app/models/GiftDTO.ts
import { GiftCategoryDTO } from './CategoryDTO';
import { DonorDTO } from './DonorDTO';
import { UserDTO } from './UserDTO';

export class Gift {
  idGift: number = 0;
  name: string = '';
  description?: string = '';
  categoryId: number = 0;
  category?: GiftCategoryDTO;
  amount: number = 0; 
  image?: string = '';
  idDonor: number = 0;
  donor?: DonorDTO;
  isDrawn: boolean = false;
  idUser?: number;
  user?: UserDTO;
  price: number = 0;
  ordersGifts: any[] = [];
  
  // השדה החדש שישמש את ה-UI בלבד
  customerQuantity: number = 0; 

  constructor(init?: Partial<Gift>) {
    Object.assign(this, init);
    // מוודא שהערך מאותחל ל-0 בבנייה
    this.customerQuantity = this.customerQuantity || 0;
  }
}

// שאר ה-DTOs נשארים ללא שינוי
export class GiftDTO {
  name: string = '';
  description?: string = '';
  categoryId: number = 0;
  quantity: number = 0;
  image?: string = '';
  idDonor: number = 0;
  price: number = 0;
  constructor(init?: Partial<GiftDTO>) { Object.assign(this, init); }
}

export class GiftNewDTO {
  idGift: number = 0;
  name: string = '';
  description: string = '';
  price: number = 0;
  numOfBuyers: number = 0;
  constructor(init?: Partial<GiftNewDTO>) { Object.assign(this, init); }
}

File: ChineseAuctionAPI/ChineseAuctionAPI/DTOs/GiftDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ChineseAuctionAPI.DTOs
{
    public class GiftDTO
    {
        [Required]
        public string Name { get; set; }
        [StringLength(500)]
        public string? Description { get; set; }
        [Required]
        public int CategoryId { get; set; }
        [Required, Range(1, 500)]
        public int Quantity { get; set; }

        public string? Image { get; set; }
        [Required]
        public int IdDonor { get; set; }
        [Required, Range(1, 1000)]
        public int Price { get; set; }
    }
    public class GiftNewDTO
    {
        public int IdGift { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Price { get; set; }
        public int NumOfBuyers { get; set; }
    }
    public class GiftUpdateDto
    {
        public int UserId { get; set; }
        public int GiftId { get; set; }
        public int Amount { get; set; }
    }
}
